analyzer/src/testFixtures/kotlin/Extensions.kt

```diff
@@ -97,3 +102,15 @@ private fun Project.filterReferencedPackages(allPackages: Set<Package>): Set<Pac
 fun ProjectAnalyzerResult.withInvariantIssues() =
     // Account for different NPM versions to return issues in different order.
     copy(issues = issues.sortedBy { it.message }.map { it.copy(timestamp = Instant.EPOCH) })
+
+fun Spec.analyze(
```

Comment:
> Why is this an extension function on Spec as that class context doe snot seem to be required?

To not pollute other scopes with this function where it's not needed.

> And maybe for clarity name this analyzeWithResolvedScopes()?

I prefer to keep the shorter name. As soon as a caller is introduces which doesn't need resolved scopes I'd foresee anyway to add a parameter like e.g.: `analyze(resolveScopes = false)`

Reply:
> As soon as a caller is introduced which doesn't need resolved scopes I'd foresee anyway to add a parameter like e.g.: `analyze(resolveScopes = false)`

I was thinking about that, too, and actually was wondering whether we should move `.withResolvedScope()` into that function in the first place, or leave it to the caller. Amount-of-code-wise that would make no difference to using a `resolveScopes = false` argument later. What do you think?

Reply:
In the more frequent case where `resolveScopes = true`, the parameter can be omit, because `true` would be the default value. I still prefer to move it into the function because **all** callers want the scopes to be resolved.

Reply:
> To not pollute other scopes with this function where it's not needed. (And to get the permission to put it into this file)

The function is only available to test code anyway, so I'm not sure what kind of "leak" you are trying to prevent here.

Reply:
We could put it into a new `Utils.kt` inside `testFixtures` or so. But if the name is supposed to stay at the generic `analyze`, then I'd indeed prefer to "hide" it inside the `Spec` scope, even if that feel like cheating just to put it into `Extensions.kt`. If needed, we can anyway move this later.